### ISSUE TITLE
How do I download Idle Master on Linux?

### ISSUE THREAD

**Issue by MiiDVeD** [jshackles/idle_master_py#18]
I'm new to Linux and I have literally no idea how to download this program? can I get some instructions? keep in mind that I'm pretty much a Linux noob.

Thx in advance :)

**Comment by ghost** [comment]
well i don't know if is the only way to download, but you can get by downloading the entire idle master py https://github.com/jshackles/idle_master_py/archive/master.zip

after, find the linux version (releases/linux) and copy to your user folder (/home/USER)

to run you will need install some python packeges, just open your terminal (ctrl+alt+t) and type this:

sudo apt-get install python-bs4 python-requests python-pil.imagetk python-colorama -y && sudo apt-get update

it will ask your linux password, no danger.

after this, by terminal open the idle file in the linux folder (cd Linux) and type python start.py
if you see the date, time and "WELCOME TO IDLE MASTER" it's working (press Enter to close the python program)

now, put your steam account in the python version is quite hard so this video may help (starts on 1:30)
https://www.youtube.com/watch?v=WC1S5dgyTdw

all done, just type "python.py" on the terminal again and if everything works the cards will start to drop after 2 hours or more (if it's working a window will pop out with one of your games image, you can close that but keep the terminal open)

sorry for the text wall and my english as well but this was the best I could do, I hope this help you and if you have any problem just ask me

ps: I'm not a part of this project or a python programer so some things are probably more complicate than should be

**Comment by MiiDVeD** [comment]
Thank you so much it works perfectly

**Comment by Pharmasolin** [comment]
Please do not close this issue

**Comment by Huskynarr** [comment]
Can you make from this an easy use tutorial on Readme.me ? 
Non Pro-User are need so what as simple way.

**Comment by aaronfranke** [comment]
Please update your guide.

**Comment by rodgons** [comment]
I'M the same guy that posted the tutorial (I had issues with the old account) now I'm very busy but I'll try update this install in the readme file of idle

UPDATE: Just sent a pull request to the main idle but you can see the guide [here](https://github.com/klounix/idle_master_py/blob/master/Releases/Linux/Linux-readme.txt)

**Comment by santiagosisto** [comment]
Thanks for your answer. 

I put up a little summary of the things I did to get it to work (with help from the internet)
So here's what I did:

1- Download idle master python (clic on clone or download):
https://github.com/jshackles/idle_master_py
2- Extract the zip then browse to Releases/Linux
3- Configure settings.txt as shown in this video(https://www.youtube.com/watch?v=r-jeMjxrorM) or "Setup Instructions - Start Here.pdf" under Documentation
4- Install needed libraries (Ubuntu example):
sudo apt-get install python-bs4 python-requests python-pil.imagetk python-colorama -y && sudo apt-get update
5- Run the app:
python2 start.py﻿

**Comment by TheJalma** [comment]
Hi! Do I need the steam client installed?
I followed your guide and successfully run the program on my Raspberry Pi 3 with Raspbian, but it shows a error about steam library:

[ 03/30/2017 02:49:38 PM ] WELCOME TO IDLE MASTER
[ 03/30/2017 02:49:38 PM ] Finding games that have card drops remaining
[ 03/30/2017 02:49:42 PM ] Reading badge page, please wait
[ 03/30/2017 02:49:42 PM ] No games have been blacklisted
[ 03/30/2017 02:49:42 PM ] Idle Master needs to idle 25 games
[ 03/30/2017 02:49:42 PM ] Starting game Motorsport Manager to idle cards
[ 03/30/2017 02:49:43 PM ] Motorsport Manager has 1 card drops remaining
[ 03/30/2017 02:49:43 PM ] Sleeping for 5 minutes
Loading Linux 32bit library
Couldn't initialize Steam API

Is that ok? Do I need to keep the browser open in my steam page while dropping the cards?

**Comment by santiagosisto** [comment]
Yes, you need your steam client installed and open. There is a linux version of the client.

You don't need to keep your browser open.

**Comment by TheJalma** [comment]
Thanks! I was able to install the steam on my RP3, but I got an error when it initialise. I tried to install libc but did get any success. 
![steam rp3 error](https://cloud.githubusercontent.com/assets/26799983/24530367/e11afccc-1587-11e7-938c-39296d418e10.png)

**Comment by puuuq** [comment]
You can't run Steam on any of the Raspberries, Steam isn't coded to work on an ARM processor.

**Comment by TheJalma** [comment]
So I'm  not able to idle my cards on a raspberry? ;/

**Comment by not-wlan** [comment]
Unless you emulate a x86(_64) Architecture or write the Steam API for ARM yourself i'd doubt it

**Comment by TheJalma** [comment]
Thanks anyway 👍

**Comment by TheJalma** [comment]
I managed to use ASF with mono this week and get my cards :)

**Comment by 97997** [comment]
Is working, thanks.

**Comment by phrogg** [comment]
I can only get the **SteamLoginSecure** token, not the **SteamLogin** token. Can someone help me please?

**Comment by phrogg** [comment]
> I can only get the **SteamLoginSecure** token, not the **SteamLogin** token. Can someone help me please?

Is it because of the steam 2FA?

**Comment by Takohashi** [comment]
> I can only get the SteamLoginSecure token, not the SteamLogin token. Can someone help me please?

steamLogin = steamRememberLogin

**Comment by phrogg** [comment]
I don't see that. Aswell.

**Comment by Takohashi** [comment]
What don't you see?

**Comment by phrogg** [comment]
Like I said I can only see **SteamLoginSecure** and **sessionId** nothing else.
